tests/testthat/test-fit.R

```diff
@@ -70,15 +70,17 @@ test_that("fit_single_optimizer gives error messages", {
     fit_mmrm(formula, fev_data),
     paste(
       "Covariance structure must be specified in formula.",
-      "Possible covariance structures include: us, toep, toeph, ar1, ar1h, ad, adh, cs, csh"
+      "Possible covariance structures include:",
+      "us, toep, toeph, ar1, ar1h, ad, adh, cs, csh, sp_exp"
     ),
     fixed = TRUE
   )
   expect_error(
     fit_single_optimizer(formula, fev_data, weights = rep(1, nrow(fev_data))),
     paste(
       "Covariance structure must be specified in formula.",
-      "Possible covariance structures include: us, toep, toeph, ar1, ar1h, ad, adh, cs, csh"
+      "Possible covariance structures include:",
+      "us, toep, toeph, ar1, ar1h, ad, adh, cs, csh"
     ),
     fixed = TRUE
   )
```

Comment:
Could you add tests for:
1. Using `covariance` from `mmrm()` (highest level user interface)
2. Using both a `covariance` argument and a formula specification (to see if this fails as expected)